service/src/main/java/tds/config/repositories/impl/ConfigRepositoryImpl.java

```diff
@@ -58,4 +65,128 @@ public ConfigRepositoryImpl(NamedParameterJdbcTemplate jdbcTemplate) {
 
         return clientSystemFlags;
     }
+
+    @Override
+    public Optional<ClientSystemMessage> findClientSystemMessage(String clientName, String messageKey, String language, String clientDefaultLanguage, String context, String subject, String grade) {
+        List<String> langauges = Arrays.asList(language, clientDefaultLanguage);
+        List<String> grades = new ArrayList<>(Arrays.asList(MATCH_ANY));
+        List<String> subjects = new ArrayList<>(Arrays.asList(MATCH_ANY));
+
+        if (grade != null && !grade.equals(MATCH_ANY)) {
```

Comment:
You're call but you could add [Apache commons lang](https://mvnrepository.com/artifact/org.apache.commons/commons-lang3/3.5) to the pom.xml if not already there and do `StringUtils.isEquals(MATCH_ANY)` which is null safe.  Same for the subject check below.